metabase/metabase
# Restore tab during navigating back to the dashboard

Related to https://github.com/metabase/metabase/issues/31786

### Description

https://github.com/metabase/metabase/pull/32150 covers main logic for preserving tab, this PR only optimizes unit tests and adds e2e

Restore tab when navigate back to the dashboard

| Before | After |
|--------|--------|
|<img width="1096" alt="Screenshot 2023-07-05 at 17 46 09" src="https://github.com/metabase/metabase/assets/125459446/a7344250-eb4a-4dca-9644-5be8891e75f0"> | <img width="1094" alt="Screenshot 2023-07-05 at 17 47 29" src="https://github.com/metabase/metabase/assets/125459446/a0b22240-4df2-4e90-906d-beeb680fca35"> | 



### How to verify

- Open a dashboard
- select not default tab (e.g. 3)
- navigate to the question from the dashboard
- click "back button" and navigate back to the dashboard
- make sure prev tab is selected

### Demo

https://www.loom.com/share/4ba8f5eee3ee463f841fd4a871635bc5

### Checklist
